addons/sale/models/sale_order_line.py

```diff
@@ -659,7 +659,9 @@ def _compute_name(self):
         for line in self:
             if not line.product_id:
                 continue
-            line.name = line.with_context(lang=line.order_partner_id.lang)._get_sale_order_line_multiline_description_sale()
+            line.name = line._get_sale_order_line_multiline_description_sale() \
+                if line.order_partner_id.is_public \
+                else line.with_context(lang=line.order_partner_id.lang)._get_sale_order_line_multiline_description_sale()
```

Comment:
I think it would be more readable if you did it on several lines, eg.

```py
if not line.order_partner_id.is_public:
    line.with_context(lang=line.order_partner_id.lang)
```